nixos/modules/profiles/macos-builder.nix

```diff
@@ -93,7 +93,12 @@ in
         };
       });
 
-  system.stateVersion = "22.05";
+  system = {
+    # To prevent gratuitous rebuilds on each change to Nixpkgs
+    nixos.revision = null;
```

Comment:
Does it matter if this is null instead of a string?

https://github.com/NixOS/nixpkgs/blob/70c476bbf7e1e33d771837bab6c50fca4d73c7f0/pkgs/top-level/release.nix#L11

https://github.com/NixOS/nixpkgs/blob/e09e6d6f62314754f5ed578bd6ffa8e292a821fc/pkgs/test/nixos-functions/default.nix#L16

Reply:
Looks like it's totally valid for it to be null, but not sure if we'd prefer that or a dummy string: https://github.com/NixOS/nixpkgs/blob/8bb14f1fed95b1a9467a7e6e5a00d24682bf4803/nixos/modules/misc/version.nix#L76-L81

Reply:
According to `nix-diff`, the only change that this option affects is the output of the `nixos-version --json` command.  If you set it to `null` then the output is:

```json
{"nixosVersion":"23.05pre-git"}
```

… whereas if you were to set it to a dummy revision then the output would be:

```json
{"nixosVersion":"23.05pre-git","nixpkgsRevision":"0000000000000000000000000000000000000000"}
```

I personally feel that setting it to `null` is the cleaner solution rather than generating a sentinel value that is invalid

Reply:
`null` sgtm. It seems to be the better solution, considering the intent of the type. If it breaks something, we'd learn what to fix, which is great.